Sources/PrinceOfVersions/ResponseModels/UpdateData/UpdateInfo.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  UpdateInfo.swift
+//  PrinceOfVersions
+//
+//  Created by Jasmin Abou Aldan on 06/07/16.
+//
+//
+
+import Foundation
+
+#if canImport(UIKit)
+import UIKit
+#elseif canImport(AppKit)
+import AppKit
+#endif
+
+// MARK: - Internal configuration data -
+
+public struct UpdateInfo: Decodable {
+
+    // MARK: - Private properties -
+
+    private var ios: [ConfigurationData]?
+    private var ios2: [ConfigurationData]?
+    private var macos: [ConfigurationData]?
+    private var macos2: [ConfigurationData]?
+    private var meta: [String: AnyDecodable]?
+
+    // MARK: - Internal properties -
+
+    internal var bundle = Bundle.main
+
+    internal var configurations: [ConfigurationData]? {
+        #if os(iOS)
+        return ios ?? ios2
+        #elseif os(macOS)
+        return macos ?? macos2
+        #endif
+    }
+
+    internal var configuration: ConfigurationData? {
+
+        guard let configurations = configurations else { return nil }
+
+        return configurations.first { configuration in
+            guard
+                let requiredOSVersion = configuration.requirements?.requiredOSVersion,
+                let installedOSVersion = sdkVersion
+            else { return false }
+            return installedOSVersion >= requiredOSVersion && meetsUserRequirements(configuration.requirements)
+        }
+    }
+
+    internal var sdkVersion: Version? {
+        #if os(iOS)
+        return try? Version(string: UIDevice.current.systemVersion)
+        #elseif os(macOS)
+        return Version(macVersion: ProcessInfo.processInfo.operatingSystemVersion)
+        #endif
+    }
+
+    internal var currentInstalledVersion: Version? {
+
+        guard
+            let currentVersionString = bundle.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String,
+            let currentBuildNumberString = bundle.object(forInfoDictionaryKey: "CFBundleVersion") as? String
+        else {
+            return nil
+        }
+
+        return try? Version(string: currentVersionString + "-" + currentBuildNumberString)
+    }
+
+    internal var metadata: [String: Any]? {
+
+        guard let configMeta = configuration?.meta else {
+            return meta?.mapValues { $0.value }
+        }
+
+        if let meta = meta {
+            return meta
+                .merging(configMeta, uniquingKeysWith: { (_, newValue) in newValue })
+                .mapValues { $0.value }
+        }
+
+        return configMeta
```

Comment:
Shouldn't you also do `.mapValues { $0.value }` here?
You are doing that to `configMeta` in the second part.

It might be a lot eaiser to do something like:
```
if meta == nil && configuration?.meta == nil { return }
let meta = meta ?? [:]
let configMeta = configuration?.meta ?? [:]

return meta
                .merging(configMeta, uniquingKeysWith: { (_, newValue) in newValue })
                .mapValues { $0.value }
```